src/doc/rustc/src/command-line-arguments.md

```diff
@@ -58,8 +58,8 @@ Example: `-l static:+whole-archive=mylib`.
 
 The kind of library and the modifiers can also be specified in a [`#[link]`
 attribute][link-attribute]. If the kind is not specified in the `link`
-attribute or on the command-line, it will link a dynamic library if available,
-otherwise it will use a static library. If the kind is specified on the
+attribute or on the command-line, it will link a dynamic library by default,
+except when building a static executable. If the kind is specified on the
```

Comment:
The previous wording was right in the sense that `-Bdynamic` with ld-like linkers is only a *preference* for a dynamic library, not a requirement, so a static library can be linked even despite `kind=dylib` if there's no dynamic version of it.

That's what I called "too platform specific" above.
That's why I also suggested to document things in terms of the inferred `kind` and not in terms of actually linked libraries.

Reply:
Based on my experiments, this isn't actually a preference: the linker will not search for static libraries if -Bdynamic is specified.